refactor: use `ElementIdGenerator` for id generation in benchmarks

EastSun5566 opened the pull request:
Closed #16 . Also bumped the Actions version and added build and benchmark tests.

EastSun5566 committed:
refactor: use `ElementIdGenerator` for id generation in benchmarks

EastSun5566 committed:
ci: update actions to use latest versions and add build with benchmarks test

mweidner037 reviewed (changes requested):
Thank you, and sorry for the delay! One small request, and then I will regenerate `benchmark_results.md` on the same machine as before.

EastSun5566 committed:
fix: keep the `nextBunchId` function and pass it to `ElementIdGenerator`

EastSun5566 committed:
chore: fix format

EastSun5566 commented on benchmarks/insert_after_custom.ts:
```diff
@@ -23,14 +23,7 @@ export async function insertAfterCustom() {
-    // but the closest comparison to existing CRDT / list-positions benchmarks.
-    return replicaId + replicaCounter++;
-  }
+  const idGenerator = new ElementIdGenerator(() => uuidv4());
```
Hey man, Happy New Year! I fixed it by reusing the `nextBunchId` function and passing it to `ElementIdGenerator`.

mweidner037 committed:
rerun benchmarks
Looks like this made sending faster due to the internal counter state instead of calling maxCounter. Though I'm not sure why saving and loading is also faster - laptop variability?
